src/algorithms/highlevel/crosssimilaritymatrix.cpp

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) 2006-2016  Music Technology Group - Universitat Pompeu Fabra
+ *
+ * This file is part of Essentia
+ *
+ * Essentia is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Affero General Public License as published by the Free
+ * Software Foundation (FSF), either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the Affero GNU General Public License
+ * version 3 along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+#include "crosssimilaritymatrix.h"
+#include "essentiamath.h"
+#include <vector>
+#include <iostream>
+#include <string>
+#include <algorithm>
+#include <functional>
+
+namespace essentia {
+namespace standard {
+
+const char* CrossSimilarityMatrix::name = "CrossSimilarityMatrix";
+const char* CrossSimilarityMatrix::category = "Music Similarity";
+const char* CrossSimilarityMatrix::description = DOC("This algorithm computes a euclidean cross-similarity matrix of two sequences of frame features. Similarity values can be optionally binarized\n\n"
+"Use default parameter values for best results while computing cross-similarity using the binarize method.\n\n"
+"The input feature arrays should be in the shape (x, y), where 'x' is the number of frames.\n\n"
+"An exception is also thrown if either one of the input audio feature arrays are empty or if the output similarity matrix is empty.\n\n"
+"References:\n"
+"[1] Serra, J., Serra, X., & Andrzejak, R. G. (2009). Cross recurrence quantification for cover song identification.New Journal of Physics.\n\n");
+
+
+void CrossSimilarityMatrix::configure() {
+  // configure parameters
+  _frameStackStride = parameter("frameStackStride").toInt();
+  _frameStackSize = parameter("frameStackSize").toInt();
+  _kappa = parameter("kappa").toReal();
+  _toBinary = parameter("toBinary").toBool();
+}
+
+// Construct a 'stacked-frames' feature vector from an input audio feature vector by given 'frameStackSize' and 'frameStackStride'
+std::vector<std::vector<Real> > CrossSimilarityMatrix::stackFrames(std::vector<std::vector<Real> >& frames, int frameStackSize, int frameStackStride) const {
+
+  if (frameStackSize == 1) {
+    return frames;
+  }
+  size_t stopIdx;
+  int increment = frameStackSize * frameStackStride;
+  std::vector<std::vector<Real> > stackedFrames;
+  stackedFrames.reserve(frames.size() - increment);
```

Comment:
You know the vector size for the stacked-frame feature, so use it for reserve too: https://stackoverflow.com/a/18992266

Reply:
Yes, ```line::57``` does it. Isn't that what you meant?